### PR TITLE
Добро пожаловать, или посторонним вход воспрещён (часть 2)

### PR DESCRIPTION
**Pull request by olegsorokin**
---
:mortar_board: [Добро пожаловать, или посторонним вход воспрещён (часть 2)](https://up.htmlacademy.ru/react/9/user/30818/tasks/12)

### CODE REVIEW

File: project/src/utils/common.ts [hunk 1]
```diff
@@ -17,3 +19,20 @@ export function formatDuration(minutes: number): string {
...
+};
+
+export function loadData<T>(state: LoadingState<T>, type: PromiseState): LoadingState<T> {
+  switch (type) {
+    case PromiseState.PENDING:
+      return { ...state, isLoading: true, isLoaded: false };
+    case PromiseState.FULFILLED:
+      return { ...state, isLoading: false, isLoaded: true };
+    case PromiseState.REJECTED:
+      return { ...state, isLoading: false, isLoaded: true };
```
Comment (dianatsevan) [inline project/src/utils/common.ts:36]: `isLoaded` should be `true` as well even if the result is failure

File: project/src/store/reducer.ts [hunk 2]
```diff
@@ -57,31 +51,26 @@ const initialState: InitialState = {
...
     data: [],
     isLoading: false,
     isLoaded: false,
-    error: null,
   },
   promo: {
     data: null,
     isLoading: false,
     isLoaded: false,
-    error: null,
```
Comment (dianatsevan) [inline project/src/store/reducer.ts:39]: Why `error` was removed?

Comment (olegsorokin) [reply project/src/store/reducer.ts:39]: In the lessons, we handle errors in `handleError` function into `api-actions`.

File: project/src/utils/common.ts (hunk 1 above)
Comment (olegsorokin) [reply project/src/utils/common.ts:36]: If this is the question, I changed to `true`, after seeing the article:
https://dev.to/sunil12738/using-request-success-failure-pattern-in-redux-to-handle-async-actions-4d0

image

Comment (olegsorokin) [reply project/src/utils/common.ts:36]: Data is loaded, but has error. Maybe.

File: project/src/store/reducer.ts [hunk 3]
```diff
@@ -15,12 +24,30 @@ type InitialState = {
...
     data: Film | null,
     isLoading: boolean,
     isLoaded: boolean,
     error: TError | null,
   },
+  film: {
+    data: Film | null,
+    isLoading: boolean,
+    isLoaded: boolean,
+    error?: TError | null,
```
[the remark this answers is not in the document]
Comment (olegsorokin) [reply project/src/store/reducer.ts]: I removed `error`.

**Pull request merged**